CMakeLists.txt

```diff
@@ -103,9 +103,8 @@
 # 2023-01-15 Carlo added C99 as the minimum required
 # 2023-08-06 PH added support for setting variable length lookbehind maximum
 
-# Increased minimum to 3.5 to workaround deprecated backward compatibility
-# since 3.27.
-cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
+# Increased minimum to 3.15 to allow use of generator expressions.
+cmake_minimum_required(VERSION 3.15 FATAL_ERROR)
```

Comment:
I used `string(REPEAT ...` which apparently requires 3.15.

Versions of Ubuntu older than 20.04 are in "extended security maintenance". I can't see users of these platforms (>5 years old) needing to use the latest-and-greatest PCRE2; they can always upgrade their CMake from the Kitware website if needed.

I'm happy to remove the `string(REPEAT ...)` line and go back down to 3.12 if you prefer, which is a 6-year-old version.

Reply:
> I'm happy to remove the `string(REPEAT ...)` line and go back down to 3.12 if you prefer

It would be also nice to update the "perl" `dev.yml` CI job to use this generator too so we have  way to prevent any future regressions.

Reply:
Hmm. That's using `cmake` from the Ubuntu repositories. We can set the build back to using the `ubuntu-20.04` GitHub runner, which would be testing with CMake 3.16.

Or we can go lower-level and really force it to use 3.12 by installing the tarball manually from KitWare.

Do we _really_ need to work with 3.12? We should be covered even on pretty old platforms by 3.15/16.

Reply:
> That's using `cmake` from the Ubuntu repositories

Not really, it is running in a Docker that is based on Debian.  Debian 10 came with cmake 3.13, but I like your idea of installing our own to make sure.

I have to admit I was more concerned about regressing the multi-config fixes for PCRE2_DEBUG though.

> Do we _really_ need to work with 3.12?

no, but we need to make sure that whatever we said we support actually works.

Reply:
Oh I see, it's using the `perl:devel` Docker container. That appears to be currently based on Debian bookworm (12) which includes CMake 3.25.

The `perl:devel-bullseye` container uses Debian bookwork (11) which bundles CMake 3.18.

I'll install an older CMake version from KitWare, to ensure we're supporting all the older targets.